src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -8,97 +8,90 @@
     private Node<T> last;
 
     private static class Node<T> {
-        T item;
-        Node<T> next;
-        Node<T> prev;
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
 
-        Node(Node<T> prev, T element, Node<T> next) {
+        private Node(Node<T> prev, T element, Node<T> next) {
             this.item = element;
             this.next = next;
             this.prev = prev;
         }
     }
 
+    @Override
     public void add(T value) {
-        linkLast(value);
+        Node<T> wasLast = last;
+        Node<T> newNode = new Node<>(wasLast, value, null);
+        last = newNode;
+        if (wasLast == null) {
+            first = newNode;
+        } else {
+            wasLast.next = newNode;
+        }
+        size++;
     }
 
+    @Override
     public void add(T value, int index) {
         checkPositionIndex(index);
         if (index == size) {
-            linkLast(value);
+            add(value);
         } else {
-            linkBefore(value, node(index));
+            insertNode(value, getNode(index));
         }
     }
 
+    @Override
     public void addAll(List<T> list) {
         for (T temp : list) {
             add(temp);
         }
     }
 
+    @Override
     public T get(int index) {
         checkElementIndex(index);
-        return node(index).item;
+        return getNode(index).item;
     }
 
+    @Override
     public void set(T value, int index) {
         checkElementIndex(index);
-        node(index).item = value;
+        getNode(index).item = value;
     }
 
+    @Override
     public T remove(int index) {
         checkElementIndex(index);
-        return unlink(node(index));
+        return removeNode(getNode(index));
     }
 
-    public T remove(T t) {
-        if (t == null) {
-            for (Node<T> x = first; x != null; x = x.next) {
-                if (x.item == null) {
-                    unlink(x);
-                    return null;
-                }
-            }
-        } else {
-            for (Node<T> x = first; x != null; x = x.next) {
-                if (t.equals(x.item)) {
-                    return unlink(x);
-                }
+    @Override
+    public T remove(T value) {
+        for (Node<T> node = first; node != null; node = node.next) {
+            if ((value == null && node.item == null) || node.item.equals(value)) {
```

Comment:
(value == node.item) || node.item.equals(value)

Reply:
Got it)